packages/@cdktf/provider-schema/src/cache.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { ConstructsMakerTarget, logger } from "@cdktf/commons";
+import * as fs from "fs-extra";
+
+// We keep this very simple since the caching feature is experimental
+// we might need to do housekeeping / include terraform / cdktf version in the future
+function cacheKey(input: ConstructsMakerTarget): string {
+  return `${encodeURIComponent(input.fqn)}@${input.version}`;
```

Comment:
I think we need to encode the version as well, as it may contain e.g. `>=5.0.0` and e.g. Windows does not allow the `>` in file names.

Reply:
I know this is experimental, but I think it's easy to improve and might save us a headache in case we take this out of experimental later (since we probably won't test this weird case on windows)